### PR TITLE
Expose additional MG params plus C interface compilation test

### PR DESCRIPTION
**Pull request by weinbe2**
This PR:
* Exposes two additional parameters to the MILC HISQ MG interface: (1) coarse link/near-null precision (half or single instead of hard-coded half), and (2) coarsening the full or preconditioned operator on intermediate and coarsest levels (instead of hard-coded preconditioned)
* Adds @mathiaswagner 's C interface compile test
* Fixes an issue in the C interface introduced by having a `bool` in the signature of the BLAS interface.

Closes #1079 and #926

### CODE REVIEW

**Comment by weinbe2** [comment]
@mathiaswagner your C test appears to have angered the gods of `clang` testing:

```
[185/252] Building C object tests/CMakeFiles/c_interface_test.dir/c_interface_test.c.o
FAILED: tests/CMakeFiles/c_interface_test.dir/c_interface_test.c.o 
/usr/bin/cc -DBUILD_MILC_INTERFACE -DBUILD_QDP_INTERFACE -DGPU_CLOVER_DIRAC -DGPU_CLOVER_HASENBUSCH_TWIST -DGPU_DOMAIN_WALL_DIRAC -DGPU_FATLINK -DGPU_GAUGE_TOOLS -DGPU_NDEG_TWISTED_MASS_DIRAC -DGPU_STAGGERED_DIRAC -DGPU_TWISTED_CLOVER_DIRAC -DGPU_TWISTED_MASS_DIRAC -DGPU_UNITARIZE -DGPU_WILSON_DIRAC -DMPI_COMMS -DMULTI_GPU -DQUDA_PRECISION=14 -DQUDA_RECONSTRUCT=7 -Itests -I../tests -I../tests/utils -I../tests/host_reference -I../lib/../include -Iinclude -isystem ../tests/googletest/include -isystem ../tests/googletest -isystem externals/eigen/src/Eigen -isystem /usr/local/cuda/include -isystem /usr/local/openmpi/include -Wall -O3 -Werror -Wno-error=unused-private-field -pthread -MD -MT tests/CMakeFiles/c_interface_test.dir/c_interface_test.c.o -MF tests/CMakeFiles/c_interface_test.dir/c_interface_test.c.o.d -o tests/CMakeFiles/c_interface_test.dir/c_interface_test.c.o -c ../tests/c_interface_test.c
cc1: error: '-Wno-error=unused-private-field': no option '-Wunused-private-field'
```

**Comment by weinbe2** [comment]
My guess is you have to add `-DCMAKE_C_COMPILER=clang` or something like that along with the other flags...

**Comment by mathiaswagner** [comment]
I cleaned up the CMAKE_C_FLAGS to be consistent with CMAKE_CXX_FLAGS and also made sure to pass CMAKE_C_COMPILER in Jenkins when we also use CMAKE_CXX_COMPILER.

**Comment by weinbe2** [comment]
> I cleaned up the CMAKE_C_FLAGS to be consistent with CMAKE_CXX_FLAGS and also made sure to pass CMAKE_C_COMPILER in Jenkins when we also use CMAKE_CXX_COMPILER.

Thanks for the quick turnover @mathiaswagner !

File: tests/CMakeLists.txt
Comment (maddyscientist) [inline tests/CMakeLists.txt]: Space inserted before `#`

File: lib/milc_interface.cpp
Comment (maddyscientist) [inline lib/milc_interface.cpp]: This appears that there's no way to enable debug verbosity.  This might be good to add.

**Comment by maddyscientist** [comment]
Apart from my minor comments, this PR looks good to go.

**Review by maddyscientist — APPROVED** [review]

**Comment by weinbe2** [comment]
Thanks for the comments @maddyscientist --- I can merge in `develop` after merging in the `computeCoarseClover` PR and address them in one go.

File: tests/CMakeLists.txt
Comment (weinbe2) [reply tests/CMakeLists.txt]: Is there a particular reason that's a problem in the context of `CMakeLists.txt`, since it's a comment (and thus should arguably match indentation), not a `pragma` in C++?

File: lib/milc_interface.cpp
Comment (weinbe2) [reply lib/milc_interface.cpp]: Good call---I added the full battery of silent/summary/verbose/debug, default summary, so we have it!

File: tests/CMakeLists.txt
Comment (weinbe2) [reply tests/CMakeLists.txt]: We indent comments elsewhere in the file, so I'm going to opt to leave it as-is.

Comment (maddyscientist) [reply tests/CMakeLists.txt]: My point was this is an unnecessary change made in this PR.  We should be striving to minimizing unnecessary diffs.

Comment (weinbe2) [reply tests/CMakeLists.txt]: Ah, fair. It looks like it got picked up because it was neighboring an addition Mathias made. I'll revert it.

Comment (weinbe2) [reply tests/CMakeLists.txt]: Actually, it looks like the issue is more that the two lines above got indented for no good reason.

Comment (weinbe2) [reply tests/CMakeLists.txt]: Fixed

**Comment by weinbe2** [comment]
This will be good to go once the build tests complete!

**Pull request merged**
